src/main/java/network/brightspots/rcv/StreamingCVRReader.java

```diff
@@ -100,9 +100,11 @@
     // to keep our code simple, we convert 1-indexed user-supplied values to 0-indexed here
     this.firstVoteColumnIndex = source.getFirstVoteColumnIndex() - 1;
     this.firstVoteRowIndex = source.getFirstVoteRowIndex() - 1;
-    this.idColumnIndex = source.getIdColumnIndex() != null ? source.getIdColumnIndex() - 1 : null;
+    this.idColumnIndex = source.getIdColumnIndex() != null && !source.getIdColumnIndex().isBlank() ?
+        Integer.parseInt(source.getIdColumnIndex()) - 1 : null;
     this.precinctColumnIndex =
-        source.getPrecinctColumnIndex() != null ? source.getPrecinctColumnIndex() - 1 : null;
+        config.isTabulateByPrecinctEnabled() ? Integer.parseInt(source.getPrecinctColumnIndex()) - 1
```

Comment:
The validation was laid out otherwise... I guess I'll change that to always complain if there's bad data in precinctColumnId even if tabulateByPrecinct isn't enabled, and revert here.